dali/driver/hid.py

```diff
@@ -207,14 +207,22 @@ async def send(self, command, in_transaction=False, exceptions=None):
             if not in_transaction:
                 self.transaction_lock.release()
 
-    async def run_sequence(self, seq, progress=None):
+    async def run_sequence(self, cmd_seq, progress=None):
+        """Run a command sequence as a transaction
+        """
+        if hasattr(cmd_seq, 'run'):
+            seq = cmd_seq.run()
+        else:
+            seq = cmd_seq
         await self.transaction_lock.acquire()
         response = None
         try:
             while True:
                 try:
                     cmd = seq.send(response)
                 except StopIteration:
+                    if hasattr(cmd_seq, 'result'):
+                        return cmd_seq.result
```

Comment:
Just `return seq`. That way the caller (who should know what they're doing) can deal with the result of the sequence.

Reply:
There is no point in returning seq if it is simply the thing that was passed in.  The purpose of this check is so that callers can write `device_types = d.run_sequence(QueryDeviceTypes(addr))` instead of having to write:
```python
seq = QueryDeviceTypes(addr)
d.run_sequence(seq)
device_types = seq.result
```

Reply:
The caller would just write `device_types = d.run_sequence(QueryDeviceTypes(addr).result`. However, that way the driver does not have to deal with API changes in a sequence.

Reply:
I'm trying to nail down the sequences API in the discussion in this pull request.  I don't see how making the caller write `device_types = d.run_sequence(QueryDeviceTypes(addr)).result` each time they want the result of a sequence is an improvement over them writing `device_types = d.run_sequence(QueryDeviceTypes(addr))`

Reply:
My thinking is that once a sequence does something where the "result" does not fit this pattern, the driver need not know about it as it is only concerned with returning the complete sequence with all its state. In the end it's really just fine either way.